File: SDKs/googleapis.com/ondemandscanning/v1/typescript/src/sdk/models/shared/slsarecipe.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";


// SlsaRecipe
/** 
 * Steps taken to build the artifact. For a TaskRun, typically each container corresponds to one step in the recipe.
**/
export class SlsaRecipe extends SpeakeasyBase {
  @Metadata({ data: "json, name=arguments" })
  arguments?: Map<string, any>;

  @Metadata({ data: "json, name=definedInMaterial" })
  definedInMaterial?: string;

  @Metadata({ data: "json, name=entryPoint" })
  entryPoint?: string;

  @Metadata({ data: "json, name=environment" })
  environment?: Map<string, any>;

  @Metadata({ data: "json, name=type" })
  type?: string;
}

File: SDKs/googleapis.com/containeranalysis/v1alpha1/go/pkg/models/shared/slsarecipe.go
package shared

// SlsaRecipe
// Steps taken to build the artifact. For a TaskRun, typically each container corresponds to one step in the recipe.
type SlsaRecipe struct {
	Arguments         map[string]interface{} `json:"arguments,omitempty"`
	DefinedInMaterial *string                `json:"definedInMaterial,omitempty"`
	EntryPoint        *string                `json:"entryPoint,omitempty"`
	Environment       map[string]interface{} `json:"environment,omitempty"`
	Type              *string                `json:"type,omitempty"`
}
